src/parser/Parser.ts

```diff
@@ -2632,30 +2632,29 @@ export interface ParseOptions {
     logger?: Logger;
 }
 
-function createReferences(): References {
-    return {
-        assignmentStatements: [],
-        classStatements: [],
-        functionExpressions: [],
-        functionStatements: [],
-        importStatements: [],
-        libraryStatements: [],
-        namespaceStatements: [],
-        newExpressions: [],
-        propertyHints: {}
-    };
-}
-
-export interface References {
-    assignmentStatements: AssignmentStatement[];
-    classStatements: ClassStatement[];
-    functionExpressions: FunctionExpression[];
-    functionStatements: FunctionStatement[];
-    importStatements: ImportStatement[];
-    libraryStatements: LibraryStatement[];
-    namespaceStatements: NamespaceStatement[];
-    newExpressions: NewExpression[];
-    propertyHints: Record<string, string>;
+export class References {
+    public assignmentStatements = [] as AssignmentStatement[];
+    public classStatements = [] as ClassStatement[];
+    public functionExpressions = [] as FunctionExpression[];
+    public functionStatements = [] as FunctionStatement[];
+    /**
+     * A map of function statements, indexed by fully-namespaced lower function name.
+     */
+    public get functionStatementLookup() {
+        if (!this._functionStatementLookup) {
```

Comment:
Compute this on-demand, so we don't pay the price until we need it. (currently only used during code actions)